src/parser/jobs/blm/modules/Procs.js

```diff
@@ -118,103 +100,106 @@ export default class Procs extends Module {
 			}))
 			this._group.nestedGroups.push(groupId)
 		}
-		this._buffs[status.id] = new Item({
-			type: 'background',
-			start: timestamp - this.parser.fight.start_time,
-			group: groupId,
-			content: <img src={status.icon} alt={status.name}/>,
-		})
 	}
 
-	_onRefreshBuff(event) {
-		const statusId = event.ability.guid
-		// When the buff refreshes, reset the wear-off timestamp, and, for timeline display purposes, end the previous buff and start a new one
-		if (statusId === STATUSES.FIRESTARTER.id) {
-			this._firestarterWears = event.timestamp + FIRESTARTER_DURATION
-			this.loseBuff(event.timestamp, STATUSES.FIRESTARTER)
-			this.applyBuff(event.timestamp, STATUSES.FIRESTARTER)
-		} else if (statusId === STATUSES.THUNDERCLOUD.id) {
-			this._thundercloudWears = event.timestamp + THUNDERCLOUD_DURATION
-			this.loseBuff(event.timestamp, STATUSES.THUNDERCLOUD)
-			this.applyBuff(event.timestamp, STATUSES.THUNDERCLOUD)
-		}
+	_onRefreshProc(event) {
+		this._stopAndSave(event.ability.guid, event.timestamp)
+		this._onGainProc(event)
 	}
 
-	loseBuff(timestamp, status) {
-		const item = this._buffs[status.id]
-		// This shouldn't happen, but it do.
-		if (!item) { return }
-
-		item.end = timestamp - this.parser.fight.start_time
-		this.timeline.addItem(item)
+	// Keep track of casts we start to help look for instant casts
+	_onBeginCast(event) {
+		this._castingSpell = event.ability
 	}
 
-	_onRemoveThundercloud(event) {
-		this._thundercloud = false
-		if (event.timestamp >= this._thundercloudWears) { // If this wore off because of time, note it
-			this._droppedT3Ps++
+	// Consolidate old onCast functions into one central function
+	_onCast(event) {
+		// Skip proc checking if we had a corresponding begincast event or the begincast we recorded isn't the same as this spell (ie. cancelled a cast, used a proc)
+		if (!this._castingSpell || this._castingSpell !== event.ability) {
```

Comment:
Tiny nitpick, you may want to assign specifically the name or guid just in case data gets weird. Passing around a full object isn't really going to be a performance hit but it is kinda weird.

Reply:
I mean, unless magic happens, this will always fail.
![image](https://user-images.githubusercontent.com/534235/47469317-14084300-d84c-11e8-97fa-45eb0021a939.png)